R/step_epi_slide.R

```diff
@@ -0,0 +1,277 @@
+#' Calculate a rolling window transformation
+#'
+#' `step_epi_slide()` creates a *specification* of a recipe step
+#'   that will generate one or more new columns of derived data by "sliding"
+#'   a computation along existing data.
+#'
+#'
+#' @inheritParams step_epi_lag
+#' @param .f A function in one of the following formats:
+#'  1. An unquoted function name with no arguments, e.g., `mean`
+#'  2. A character string name of a function, e.g., `"mean"`. Note that this
+#'     can be difficult to examine for mistakes (so the misspelling `"maen"`
+#'     won't produce an error until you try to actually fit the model)
+#'  3. A base `R` lambda function, e.g., `function(x) mean(x, na.rm = TRUE)`
+#'  4. A new-style base `R` lambda function, e.g., `\(x) mean(x, na.rm = TRUE)`
+#'  5. A one-sided formula like `~ mean(.x, na.rm = TRUE)`.
+#'
+#'  Note that in cases 3 and 4, `x` can be any variable name you like (for
+#'  example `\(dog) mean(dog, na.rm = TRUE)` will work). But in case 5, the
+#'  argument must be named `.x`. A common, though very difficult to debug
+#'  error is using something like `function(x) mean`. This will not work.
+#' @param f_name a character string of at most 20 characters that describes
+#'   the function. This will be combined with `prefix` and the columns in `...`
+#'   to name the result using `{prefix}{f_name}_{column}`. It will be determined
+#'   automatically using `clean_f_name()`.
```

Comment:
question: what does this argument do that prefix doesn't?

Reply:
The prefix indicates the step name, and `f_name` the function. Imagine I applied this step twice to the same column, once with the trailing mean and then for the trailing sd. I'd want to distinguish them.

Reply:
Hm I would probably unify prefix and f_name since they seem to have overlapping functionality. This flexibility goes against "there should be one way to do it"
```r
step_epi_slide(..., prefix = "epi_slide_mean", f_name = "")
step_epi_slide(..., prefix = "", f_name = "epi_slide_mean")
step_epi_slide(..., ~mean(.x), prefix = "epi_slide")
step_epi_slide(..., mean, prefix = "epi_slide")
```
Saving the user a few characters of typing isn't worth all the extra code to me.

Reply:
So I'm just going to go ahead and make the change to requiring the user to be explicit. I think it will reduce the complexity of the code and remove buggy edge cases in the future.

Reply:
I would suggest being a bit careful. The convention in tidy models is to have new column names determined automatically by default. The difficulty here is that you want "default" to indicate both the `step_*()` that was used and to allow for the case that users are likely to use this step on the same columns but with different functions. So that's the justification for the overlapping functionality. I felt like these considerations were higher priority (following conventions, automatically working if you slide two funs on the same column).

Reply:
I spoke to David a bit and tried it without it and decided against it. This seems fine.